robottelo/config/__init__.py

```diff
@@ -16,7 +16,27 @@
     os.environ['ROBOTTELO_DIR'] = str(robottelo_root_dir)
 
 
-settings = LazySettings(
+class DelayedBox(base.DynaBox):
+    """This class allows settings.server.hostname to have like a property"""
+
+    def __getattribute__(self, name):
```

Comment:
Can you please explain why the `__getattribute__` is used? The intended behavior described in the docstring could be done with properties.

```
class DelayedBox(base.DynaBox):
    @property
    def hostname(self):
        return self._hostname

    @hostname.setter
    def set_hostname(self, value):
        self._hostname = value

Reply:
I didn't use property because I want more fine control over the get/set behavior since modified hostname must only apply under `settings.server` and not other theoretical hostname paths.
For example: `settings.dogfood.hostname`


Ultimately, this PR will almost certainly not be the correct solution for our problem. Because of this, I'll likely close this PR in the near future.